src/types/Server.ts

```diff
@@ -374,7 +374,11 @@ export enum ServerAction {
   REST_SIGNIN = 'signin',
   REST_SIGNON = 'signon',
   REST_SIGNOUT = 'signout',
-  REST_PASSWORD_RESET = 'passwordreset',
+  REST_PASSWORD_RESET = 'passwordReset',
```

Comment:
The URI naming in standard REST API is always lower case. 

Please read https://restfulapi.net/resource-naming/ and the issue on the project where a lengthy discussion happen on REST API standard design. 

After the reading, you will for example deduce that all mail related operations should be under .../mail: 

.../mail/verify
.../mail/verify/resend

And EULA related: 

.../eula
.../eula/verify 

So before you start coding, I think you have to make a proposal on the naming scheme of the REST API endpoints you want to implement to avoid having to change code in tree places ;) 

For example here, verify keyword in URI is debatable, I would really prefer to replace it by check or control. 

So from now on, the sub path in the endpoints URI does not match totally the ServerAction, so you probably have to think about data structure that permits to express: that endpoints will permit to do control on mail or control on eula.

Reply:
Ok, thanks for the informations and advices !

So the Auth API should look like this ?
/auth/signin
/auth/signon
/auth/signout
/auth/password/reset
/auth/mail/check
/auth/mail/resend
/auth/eula
/auth/eula/check

And what about ServerActions ? 
Do I only create an entity constant (e.g REST_MAIL = 'mail') and then call manually REST_MAIL + '/check' or do I create a constant for each endpoint (e.g REST_CHECK_MAIL = 'mail/check') ?

Reply:
I'm fine with your proposal for authentication endpoints that follows that kind of scheme: 

/feature/data/action

For the path, just put the action in ServerAction to begin with, not the full URI subpath with slash (never put slash in constants definition, that can lead to various parsing side effects and portability issues). And hardcode the data part.  
After a bit more work on the new API, we will see what other part of the URI can be 'constanstified'. That kind of cleanups are better targeted after coding, before, we are not sure that we not have overlooked or missed something.